### PR TITLE
Boutique - samples and default

### PR DESCRIPTION
**Pull request by klokane**
add: Implementation of rendering JSON from samples && default
fix: samples and defaults for Named Types
change: reorganize test fixtures

### CODE REVIEW

File: src/RefractAST.cc [hunk 1]
```diff
@@ -194,47 +194,75 @@ namespace drafter
...
+
+    struct TypeSectionData {
+        RefractElements defaults;
+        RefractElements samples;
+        std::vector<std::string> descriptions;
+    };
+
+    template <typename T>
     struct ExtractTypeSection
     { // This will handle primitive elements
```
Comment (klokane) [inline src/RefractAST.cc]: Remove comment

File: src/refract/RenderJSONVisitor.cc [hunk 2]
```diff
@@ -11,6 +11,8 @@
 #include "sosJSON.h"
 #include <sstream>
 
+#include <iostream>
```
Comment (klokane) [inline src/refract/RenderJSONVisitor.cc]: remove

**Comment by pksunkara** [comment]
I have reviewed the last commit. Can you remove that commit from this PR and make a separate PR for it? It's just too big :cry:

**Comment by klokane** [comment]
OK will split into smaller steps

**Comment by klokane** [comment]
@pksunkara splited into three small logical parts
see #66 #67 #68

**Comment by pksunkara** [comment]
Thanks :smile:.

**Pull request closed without merging**